vault/mount.go

```diff
@@ -206,11 +221,18 @@ func (c *Core) unmount(path string) error {
 	}
 
 	// Verify exact match of the route
-	match := c.router.MatchingMount(path)
-	if match == "" || path != match {
+	match := c.router.MatchingMountEntry(path)
+	if match == nil || path != match.Path {
 		return fmt.Errorf("no matching mount")
 	}
 
+	// Do not allow singleton mounts to be removed
```

Comment:
Since both 'system' and 'cubbyhole' types belongs to both singletons protected mounts, isn't this code unreachable, since the protectedMounts check is placed before?